File: backend/src/models/message-model.ts
import { Expose, Transform } from "class-transformer";
import { ObjectId } from "mongodb";
import { LocationModel } from "./location-model";

export class MessageModel {
  @Expose()
  conversationId: string;

  @Expose()
  senderId: string;

  @Expose()
  createdAt: Date;

  @Expose()
  text?: string;

  @Expose()
  imageSrc?: string;

  @Expose()
  location?: LocationModel;

  @Expose()
  messageType: MessageType;
}

export enum MessageType {
  TEXT = "TEXT",
  IMAGE = "IMAGE",
  LOCATION = "LOCATION",
}

File: bilfind_app/lib/models/message_model.dart
class MessageModel {
  MessageModel({
    required this.conversationId,
    required this.senderId,
    required this.createdAt,
    required this.text,
    required this.messageType,
  });
  late final String conversationId;
  late final String senderId;
  late final DateTime createdAt;
  late final String text;
  late final String messageType;

  MessageModel.fromJson(Map<String, dynamic> json) {
    conversationId = json['conversationId'];
    senderId = json['senderId'];
    createdAt = DateTime.parse(json['createdAt']).toLocal();
    text = json['text'];
    messageType = json['messageType'];
  }
}
